crates/services/txpool_v2/src/pending_pool.rs

```diff
@@ -0,0 +1,609 @@
+        outputs: Vec<Output>,
+        rng: &mut StdRng,
+    ) -> ArcPoolTx {
+        let predicate = op::ret(RegId::ONE).to_bytes().to_vec();
+        let owner = Input::predicate_owner(&predicate);
+        let mut tx = TransactionBuilder::script(vec![], vec![]);
+        tx.script_gas_limit(1000000);
+        if inputs.is_empty() {
+            tx.add_input(Input::coin_predicate(
+                UtxoId::new([0; 32].into(), 0),
+                owner,
+                outputs
+                    .iter()
+                    .map(|output| match output {
+                        Output::Coin { amount, .. } => *amount,
+                        _ => 0,
+                    })
+                    .sum(),
+                AssetId::default(),
+                Default::default(),
+                Default::default(),
+                predicate.clone(),
+                Default::default(),
+            ));
+        } else {
+            for (idx, input) in inputs.iter().enumerate() {
+                match input {
+                    MissingInput::Utxo(input) => {
+                        tx.add_input(Input::coin_predicate(
+                            *input,
+                            owner,
+                            10,
+                            AssetId::default(),
+                            Default::default(),
+                            Default::default(),
+                            predicate.clone(),
+                            Default::default(),
+                        ));
+                    }
+                    MissingInput::Contract(contract_id) => {
+                        tx.add_input(Input::contract(
+                            UtxoId::default(),
+                            Default::default(),
+                            Default::default(),
+                            Default::default(),
+                            *contract_id,
+                        ));
+
+                        tx.add_output(Output::Contract(OutputContract {
+                            input_index: idx.try_into().unwrap(),
+                            balance_root: Default::default(),
+                            state_root: Default::default(),
+                        }));
+                    }
+                }
+            }
+        }
+        for output in outputs {
+            tx.add_output(output);
+        }
+        // To differentiates the transactions
+        tx.add_random_fee_input(rng);
+        let mut tx = tx.finalize();
+        tx.estimate_predicates(
+            &CheckPredicateParams::default(),
+            MemoryInstance::new(),
+            &EmptyStorage,
+        )
+        .expect("Failed to estimate predicates");
+        let tx = tx
+            .into_checked_basic(1u32.into(), &ConsensusParameters::default())
+            .unwrap();
+        Arc::new(PoolTransaction::Script(tx, Metadata::new(1, 1, 0)))
+    }
+
+    #[test]
+    fn test_pending_pool_one_tx_one_dependent_input_utxo() {
```

Comment:
For these tests. I don't think there is any need to specify `test_pending_pool` for all of them--that's what the namespace is for.

The pending pool has 2 pub methods: `insert_transaction` and `new_known_txs`. I'd suggest putting the tests in terms of those methods. 

I'd reframe this specific test as `new_known_tx__resturns_expected_tx_when_dependent_input_provided`

That begs the question: is it possible to test `insert_transaction` without calling `new_known_tx`?

Looks like great coverage in general though.

Reply:
Yeah it's possible !

Reply:
I have renamed the tests. Tell me if it's better :)